### PR TITLE
[WIP] improving user expierence

### PR DESCRIPTION
**Pull request by AlekseySh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,6 +55,17 @@ Here are a few examples of such tasks from the computer vision sphere:
 </details>
 
 
+<details>
+<summary>Why do I need OML?</summary>
+<p>
+- If I need image embeddings I can simply train a vanilla classifier and take its penultimate layer.
+- Well, it makes sense as a starting point. But there are several drawbacks:
+todo
```
Comment (DaloroAT) [inline README.md:63]: todo

File: README.md [hunk 2]
```diff
@@ -117,31 +129,32 @@ Please, see "Minimal examples" section for more details.
...
+## Get started
+The design of OML assumes that you may train your model in 2 different ways:
 
+* **Via Configs**. The best option is if your dataset and pipeline are standard enough or if you are not
+experienced in Machine Learning or Python. You can find more details in the *examples* submodule and it's
+[Readme](https://github.com/OML-Team/open-metric-learning/blob/main/examples/).
 
-Note, if you pass one of the special keys to the constructor we will download the pretrained checkpoint for you automatically.
-However, you can also check link in `weights & config` if you want to download weights manually or to reproduce the experiment.
+* **Via Python**. The most flexible, but but knowledge-requiring approach.
```
Comment (DaloroAT) [inline README.md:151]: 2x but

File: examples/README.md [hunk 3]
```diff
@@ -88,3 +120,110 @@ are split for training and the other 11,316 (60,502 images) labels are used for
...
+show_registry()
+```
+
+## How to use my own implementation of loss, model, augmentations, etc?
+You should put your python object inside the corresponding registry by some key.
+It allows you to access this object in the config file by that key.
+
+You may change the following blocks and to work correctly some of them have to inherit our interfaces:
+* `Transforms`, `Sampler`, `Optimizer`, `Scheduler` - follow the standard PyTorch interfaces.
+* `Model` - have to be child of `IExtractor` (see `oml.interfaces.models`)
```
Comment (DaloroAT) [inline examples/README.md:171]: is child correct word?

File: oml/lightning/entrypoints/train.py [hunk 4]
```diff
@@ -72,6 +74,9 @@ def pl_train(cfg: TCfg) -> None:
     optimizer = get_optimizer_by_cfg(cfg["optimizer"], params=extractor.parameters())
     scheduler = get_scheduler_by_cfg(cfg["scheduler"], optimizer=optimizer) if cfg["scheduler"] is not None else None
 
+    assert isinstance(extractor, IExtractor), "You model must to be child of IExtractor"
```
Comment (DaloroAT) [inline oml/lightning/entrypoints/train.py:77]: check child word

File: tests/test_oml/test_registry/test_registry.py [hunk 5]
```diff
@@ -1,34 +1,23 @@
-from pathlib import Path
-from typing import Any, Callable, Dict
+# type: ignore
```
Comment (DaloroAT) [inline tests/test_oml/test_registry/test_registry.py:1]: why?

**Review by DaloroAT — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (AlekseySh) [reply README.md:151]: thx

File: tests/test_oml/test_registry/test_registry.py (hunk 5 above)
Comment (AlekseySh) [reply tests/test_oml/test_registry/test_registry.py:1]: I have to write too many "type ignores" instead 
I

File: examples/README.md (hunk 3 above)
Comment (AlekseySh) [reply examples/README.md:171]: wiki says that child is okay 
https://en.wikipedia.org/wiki/Object-oriented_programming

but I can replace it with `successor`

File: oml/lightning/entrypoints/train.py (hunk 4 above)
Comment (AlekseySh) [reply oml/lightning/entrypoints/train.py:77]: same

File: README.md (hunk 1 above)
Comment (AlekseySh) [reply README.md]: done

**Review by DaloroAT — APPROVED** [review]
